### DISCUSSION TITLE
[2024-02-05] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#8954] [Weekly Demo Day]
- James: Refactor OrderMask to avoid memory copying #8863
- Radosław: Write support for S3 #8921
- Radosław: Check type of `self` in static dispatch #8867
- Greg: Implement relational NULL/Nothing for join for in-memory tables #8849
- Greg: Attach a warning when Nothing is used in a Filter_Condition #8865
- Greg: Disable tests that fail because of locale-specific Postgres Unicode collation differences #8869
- Greg: Implement Table.replace for the in-memory backend. #8935
- Cassandra: 8830 add google analytics authentication and reporting to enso #8907
- Cassandra: Create Google Analytics Read Blog #8930 
- Cassandra: Defined full run_google_report design #8908 
- Ad: Split HashJoin to SimpleHashJoin and CompoundHashJoin #8850
- Jaroslav: Binary operator resolution based on `that` value #8779
- Jaroslav: Benchmark startup of Base_Tests and Table_Tests #8820
- Jaroslav: Argument block application has no place in type body #8877
- Jaroslav: Canonicalize the temporary directory files #8879
- Jaroslav: Log histograms with -Dorg.enso.persist.Logger.level=debug #8881
- Jaroslav: Upgrading to GraalVM 21.0.2 #8883
- Jaroslav: Using latest GraalVM bits in Enso #8885
- Jaroslav: Avoid `runtime-compiler` dependencies on Jackson #8894
- Jaroslav: How to get a Java library from Maven central #8912
- Jaroslav: Always runEngineDistribution with -ea #8915
- Jaroslav: Execute and debug `.enso` files with `bin/enso` in VSCode #8923
- Jaroslav: Store whole IR.Module in .bindings cache #8924
- Jaroslav: Better `runEngineDistribution --run test/Base_Tests "filter"` #8940
- Hubert: Don't cancel pending visualization's upserts #8853
- Hubert: Reduced resources for various Akka dispatchers #8892
- Hubert: Minor cleanups to docker image #8925
- Hubert: Further optimizations to test resources #8936
- Hubert: Disable flaky test #8967 
- Pavel: Remove fansi dependency from runtime-compiler #8847
- Pavel: Refactor Base_Tests/src/Data to Test_New #8890
- Pavel: Fix teardown in SQLite File spec #8918
- Pavel: Remove akka from downloader and from runtime #8953

**Comment by AdRiley** [comment]
![image](https://github.com/enso-org/enso/assets/1720119/ed58b40b-a446-4e8a-b1b9-8fa8c2a65ad4)
